iothub/device/src/Authentication/ClientAuthenticationWithSharedAccessPolicy.cs

```diff
@@ -49,10 +65,13 @@ public string Key
         /// <summary>
         /// Name of the shared access policy.
         /// </summary>
-        public string PolicyName
+        /// <remarks>
+        /// The name of relevant field in a connection string is "SharedAccessKeyName", so we call the property "KeyName" here.
```

Comment:
Although the auth method here is using shared access policy, this property is supposed to be "SharedAccessKeyName" within a connection string. Also, we use "SharedAccessKeyName" everywhere else, so rename it.